### PR TITLE
fix(hc): Use RpcOrganization in Pipeline

### PR DESCRIPTION
**Pull request by RyanSkonnord**
This is a step toward allowing views that depend on the auth pipeline to run in the control silo mode.

Add an HC service method for updating Organization flags, replacing a write operation done by `AuthHelper.disable_2fa_required`.

### CODE REVIEW

File: src/sentry/services/hybrid_cloud/organization/impl.py [hunk 1]
```diff
@@ -281,6 +282,18 @@ def _query_organizations(
...
 
+    def update_flags(self, *, organization_id: int, flags: RpcOrganizationFlagsUpdate) -> None:
+        updates = models.F("flags")
+        for (name, value) in flags.items():
+            if value is True:
+                updates = updates.bitor(Organization.flags[name])
+            elif value is False:
+                updates = updates.bitand(~Organization.flags[name])
+            else:
+                raise TypeError
```
Comment (markstory) [inline src/sentry/services/hybrid_cloud/organization/impl.py:293]:
```suggestion
                raise TypeError(f"Invalid value {value} received for update_flags")
```

Might be helpful for future us.

**Review by markstory — APPROVED** [review]
Looks good to me, but the test failures seem relevant.

**Review by dashed — APPROVED** [review]

File: src/sentry/services/hybrid_cloud/project/impl.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+from sentry.models import Project, ProjectOption
+from sentry.services.hybrid_cloud.project import ProjectService, RpcProject
+
+
+class DatabaseBackedProjectService(ProjectService):
+    def get_option(self, *, project: RpcProject, key: str) -> Union[str, int, bool]:
+        orm_project = Project.objects.get(id=project.id)
+        result = ProjectOption.objects.get_value(orm_project, key, None)
+        return result  # type: ignore
```
[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/services/hybrid_cloud/project/impl.py:13]: > I think the biggest drawback would be that we could no longer have a simple function with the signature

That's a good point. Having that interface stay simple is more important than optimizing away a few option reads that cross silos.

File: src/sentry/tasks/integrations/migrate_repo.py [hunk 3]
```diff
@@ -47,5 +48,5 @@ def migrate_repo(repo_id: int, integration_id: int, organization_id: int) -> Non
         from sentry.mediators.plugins import Migrator
 
         Migrator.run(
-            integration=integration, organization=Organization.objects.get(id=organization_id)
+            integration=integration, organization=organization_service.get(id=organization_id)
```
Comment (markstory) [inline src/sentry/tasks/integrations/migrate_repo.py:51]: Does the `Repository` query on line 16 of this file also need to be updated? It looks like this task will run on control silo, as it is spawned from the integration's `post_install` hook method.

File: src/sentry/integrations/bitbucket_server/integration.py [hunk 4]
```diff
@@ -305,17 +310,22 @@ class BitbucketServerIntegrationProvider(IntegrationProvider):
...
-            provider__in=["bitbucket_server", "integrations:bitbucket_server"],
-            integration_id__isnull=True,
-        ).values_list("id", flat=True)
+            providers=["bitbucket_server", "integrations:bitbucket_server"],
+            has_integration=False,
+        )
 
-        for repo_id in repo_ids:
+        for repo in repos:
             migrate_repo.apply_async(
```
Comment (markstory) [inline src/sentry/integrations/bitbucket_server/integration.py:326]: We might want to make this into an RPC/outbox task. The `migrate_repo` task does a bunch of write operations to Repository and plugin settings.

File: src/sentry/integrations/vsts/integration.py [hunk 5]
```diff
@@ -356,19 +359,19 @@ class VstsIntegrationProvider(IntegrationProvider):  # type: ignore
...
-            provider__in=["visualstudio", "integrations:vsts"],
-            integration_id__isnull=True,
-        ).values_list("id", flat=True)
+            providers=["visualstudio", "integrations:vsts"],
+            has_integration=False,
+        )
 
-        for repo_id in repo_ids:
+        for repo in repos:
             migrate_repo.apply_async(
```
Comment (markstory) [inline src/sentry/integrations/vsts/integration.py:372]: We might want to make this into an RPC/outbox message as well, similar to other repo migrations.

**Pull request merged**